include/seqan3/alignment/pairwise/detail/policy_optimum_tracker_simd.hpp

```diff
@@ -0,0 +1,235 @@
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/core/simd/simd_traits.hpp>
+#include <seqan3/core/type_traits/lazy.hpp>
+#include <seqan3/range/views/zip.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Function object that compares and updates the alignment optimum for the vectorised global alignment algorithm.
+ * \ingroup pairwise_alignment
+ *
+ * \details
+ *
+ * This operation is specific to the global alignment in vectorised mode. The tracking of the last cells of the
+ * different alignment matrices that are computed simultaneously in one vector unit depends on how the scoring of the
+ * global alignment works. Any alignment matrix that is smaller than the largest matrix defined by the longest sequence
+ * in the collection of the first sequences and in the collection of the second sequences, will have its last cell
+ * in the middle of the encompassing matrix. In order to track this cell without checking every cell of the alignment
+ * matrix for the correct coordinate, the last cell of every contained matrix is projected along its diagonal to either
+ * the last row or the last column of the encompassing matrix. Within the algorithm only the cells of the last row,
+ * respectively column are tracked. Since the cells to track are fixed, the respective coordinates for every
+ * contained matrix can be precomputed. Subsequently, within this update operation the score is only updated if the
+ * coordinate of the current cell compares equal to the precomputed coordinate of any of the contained matrices.
+ */
+struct max_score_updater_simd_global
+{
+
+    /*!\brief Compares and updates the optimal score-coordinate pair.
+     * \tparam score_t The type of the score to track; must model std::assignable_from `const & score_t`.
+     * \tparam coordinate_t The type of the coordinate to track; must be a seqan3::matrix_index type with members that
+     *                      model seqan3::simd::simd_concept.
+     *
+     * \param[in,out] optimal_score The optimal score to update.
+     * \param[in,out] optimal_coordinate The optimal to update.
+     * \param[in] current_score The score of the current cell.
+     * \param[in] current_coordinate The coordinate of the current cell.
+     *
+     * \details
+     *
+     * Compares the coordinate of the current cell with the precomputed coordinate which represent the projected
+     * last cells of the contained matrices. If a coordinate matches the precomputed one, then the respective score
+     * will be set for the optimal score.
+     */
+    template <typename score_t, typename coordinate_t>
+    //!\cond
+        requires (std::assignable_from<score_t &, score_t const &> &&
+                  requires (coordinate_t coordinate)
+                  {
+                      requires simd_concept<decltype(coordinate.col)>;
+                      requires simd_concept<decltype(coordinate.row)>;
+                  })
+    //!\endcond
+    void operator()(score_t & optimal_score,
```

Comment:
side question, but why is this function not static? This way you would need to store an instance. The updater doesn't seem to have a state

Reply:
That is the regular implementation of a function object. You don't need to have something static here. Without a state this is even an empty class that occupies at most one byte. I think that is not the problem. I could rewrite it as a named lambda object, but then the requires clause becomes unreadable because of `remove_cvref_t<decltype(auto)>` for every variable I refer to. Hence it is a function object. Second, in the semi-global alignment I need to restrict the area for which the alignment score can be tracked based on the sizes of the contained matrices. This will be done, by storing the outermost column and row coordinate for each submatrix.